pages/userprofile/index.js

```diff
@@ -0,0 +1,195 @@
+  &:hover {
+    transform: translateY(-5px);
+  }
+`;
+
+const Main = styled.main`
+  padding: 20px;
+`;
+
+const Details = styled.details`
+  margin-bottom: 20px;
+
+  & > summary {
+    cursor: pointer;
+    padding: 10px;
+    background-color: grey;
+    color: black;
+    border-radius: 5px;
+    margin-bottom: 10px;
+    transition: background-color 0.3s ease;
+
+    &:hover {
+      background-color: #0056b3;
+    }
+  }
+
+  & > h2 {
+    font-size: 1.5rem;
+    margin-bottom: 10px;
+  }
+
+  & > div {
+    margin-top: 10px;
+    text-align: center;
+  }
+`;
+
+const UserDetail = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  margin-bottom: 20px;
+`;
+
+const ImageWrapper = styled.div`
+  border-radius: 50%;
+  width: 150px;
+  height: 100px;
+  overflow: hidden;
+`;
+
+const UserPage = ({ favorites, onToggleFavorite }) => {
+  const router = useRouter();
+  const { isReady } = router;
+  const { data, mutate } = useSWR("/api/providers");
+  const { data: session, status } = useSession();
+
+  if (!data || !isReady) return <div>Loading...</div>;
+  console.log(session);
+  if (status === "unauthenticated") {
+    router.push("/");
+    return null;
+  }
+
+  const myCards = data
+    ? data.filter((card) => session.user.email === card.author)
+    : [];
+  const favoriteCards = data
+    ? data.filter((card) => favorites.includes(card._id))
+    : [];
+
+  return (
+    <Container>
+      <Header>
+        <Link href="/">
+          <Headline> &larr; Back to Categories</Headline>
+        </Link>
+      </Header>
+
+      <Main>
+        <UserDetail>
+          {session.user.image && (
+            <ImageWrapper>
+              <Image
+                src={session.user.image}
+                alt="User profile picture"
+                strategy="responsive"
+                width={200}
+                height={200}
+              />
+            </ImageWrapper>
+          )}
+          <h3>Welcome {session.user.name}</h3>
+        </UserDetail>
+        <Details>
+          <summary>Show my Cards</summary>
+          <h2>My Cards</h2>
+          <CardWrapper>
+            {myCards.length > 0 ? (
+              myCards.map((card) => (
+                <Card key={card._id}>
+                  <ServiceProvider key={card._id} card={card} />
+                </Card>
+              ))
+            ) : (
+              <div>No cards to display</div>
+            )}
+          </CardWrapper>
+        </Details>
+
+        <Details>
+          <summary>Show my Favorites</summary>
+          <h2>Favorite Cards</h2>
+          <CardWrapper>
+            {favoriteCards.length > 0 ? (
+              favoriteCards.map((card) => (
+                <Card key={card._id}>
+                  <FavoriteButton
+                    onClick={() => onToggleFavorite(card._id)}
+                    isFavorite={favorites.includes(card._id)}
+                  />
+                  <ServiceProvider key={card._id} card={card} />
```

Comment:
```suggestion
                  <ServiceProvider card={card} />
```
only the single element that is returned at the highest level of a mapping function needs to have a key prop